artemis-web/src/main/java/org/apache/activemq/artemis/component/WebServerComponent.java

```diff
@@ -145,9 +149,16 @@ public void internalStop() throws Exception {
                //tmpdir will be removed by deleteOnExit()
                //somehow when broker is stopped and restarted quickly
                //this tmpdir won't get deleted sometimes
-               boolean fileDeleted = TimeUtils.waitOnBoolean(false, 10000, tmpdir::exists);
+               boolean fileDeleted = TimeUtils.waitOnBoolean(false, 5000, tmpdir::exists);
                if (!fileDeleted) {
-                  ActiveMQWebLogger.LOGGER.tmpFileNotDeleted(tmpdir);
+                  //because the execution order of shutdown hooks are
+                  //not determined, so it's possible that the deleteOnExit
+                  //is executed after this hook, in that case we force a delete.
+                  FileUtil.deleteDirectory(tmpdir);
+                  logger.debug("Force to delete temporary file on shutdown: " + tmpdir.getAbsolutePath());
```

Comment:
I'm going to merge this,
but just as a future reference, we do a lot of if (logger.isDebugEnabled());

This is a negative test and it won't be affected, so no big deal, and it will be merged.. 


just saying that for future reference.